pkg/gitreceive/storage/bucket.go

```diff
@@ -0,0 +1,29 @@
+package storage
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+var (
+	ACLPublicRead = aws.String("public-read")
+)
+
+func BucketExists(svc *s3.S3, bucketName string) (bool, error) {
+	_, err := svc.HeadBucket(&s3.HeadBucketInput{
+		Bucket: aws.String(bucketName),
+	})
+	if err != nil {
```

Comment:
Is there a specific error that the AWS SDK returns when the API returns `404 Not Found`? If so, it would be great to only return `false, nil` in that case, and `false, err` in all other error cases.